File: StudentResultManager/student_result_manager.py
class StudentResultManager:
    def __init__(self):
        self.students = {}

    def add_student(self, name):
        if name not in self.students:
            self.students[name] = {}
            return True
        return False

    def add_marks(self, name, subject, marks):
        if name in self.students:
            self.students[name][subject] = marks
            return True
        return False

    def calculate_total(self, name):
        if name in self.students and self.students[name]:
            return sum(self.students[name].values())
        return 0

    def calculate_average(self, name):
        if name in self.students and self.students[name]:
            return self.calculate_total(name) / len(self.students[name])
        return 0

    def calculate_grade(self, average):
        if average >= 90:
            return 'A+'
        elif average >= 80:
            return 'A'
        elif average >= 70:
            return 'B'
        elif average >= 60:
            return 'C'
        elif average >= 50:
            return 'D'
        else:
            return 'F'

    def generate_report(self, name):
        if name in self.students:
            total = self.calculate_total(name)
            average = self.calculate_average(name)
            grade = self.calculate_grade(average)
            
            print(f"\n--- Student Report ---")
            print(f"Name: {name}")
            print("\nSubject-wise Marks:")
            for subject, marks in self.students[name].items():
                print(f"{subject}: {marks}")
            print(f"\nTotal Marks: {total}")
            print(f"Average: {average:.2f}")
            print(f"Grade: {grade}")
            print("----------------------")
        else:
            print("Student not found!")

def main():
    srm = StudentResultManager()
    
    while True:
        print("\nStudent Result Manager")
        print("1. Add Student")
        print("2. Add Marks")
        print("3. View Report")
        print("4. Exit")
        
        choice = input("Enter your choice (1-4): ")
        
        if choice == '1':
            name = input("Enter student name: ")
            if srm.add_student(name):
                print(f"Student {name} added successfully!")
            else:
                print("Student already exists!")
                
        elif choice == '2':
            name = input("Enter student name: ")
            if name in srm.students:
                subject = input("Enter subject name: ")
                try:
                    marks = float(input("Enter marks (0-100): "))
                    if 0 <= marks <= 100:
                        srm.add_marks(name, subject, marks)
                        print("Marks added successfully!")
                    else:
                        print("Marks should be between 0 and 100!")
                except ValueError:
                    print("Please enter a valid number!")
            else:
                print("Student not found!")
                
        elif choice == '3':
            name = input("Enter student name: ")
            srm.generate_report(name)
            
        elif choice == '4':
            print("Thank you for using Student Result Manager!")
            break
            
        else:
            print("Invalid choice! Please try again.")

if __name__ == "__main__":
    main()

File: Projects/StudentResultManager/src/main/java/com/student/result/manager/StudentResultManager.java
package com.student.result.manager;

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class StudentResultManager {
    private static List<Student> students = new ArrayList<>();
    private static Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) {
        boolean running = true;
        
        System.out.println("=== Student Result Manager ===");
        
        while (running) {
            displayMenu();
            int choice = getIntInput("Enter your choice (1-5): ", 1, 5);
            
            switch (choice) {
                case 1:
                    addStudent();
                    break;
                case 2:
                    viewAllStudents();
                    break;
                case 3:
                    searchStudent();
                    break;
                case 4:
                    System.out.println("\nThank you for using Student Result Manager!");
                    running = false;
                    break;
                default:
                    System.out.println("Invalid choice. Please try again.");
            }
        }
        
        scanner.close();
    }
    
    private static void displayMenu() {
        System.out.println("\n--- Main Menu ---");
        System.out.println("1. Add Student Result");
        System.out.println("2. View All Results");
        System.out.println("3. Search Student Result");
        System.out.println("4. Exit");
    }
    
    private static void addStudent() {
        System.out.println("\n--- Add New Student Result ---");
        
        System.out.print("Enter Roll Number: ");
        String rollNumber = scanner.nextLine();
        
        System.out.print("Enter Student Name: ");
        String name = scanner.nextLine();
        
        int[] marks = new int[Student.getSubjects().length];
        System.out.println("\nEnter marks out of " + Student.getMaxMarks() + " for each subject:");
        
        for (int i = 0; i < Student.getSubjects().length; i++) {
            marks[i] = getIntInput(Student.getSubjects()[i] + ": ", 0, Student.getMaxMarks());
        }
        
        Student student = new Student(rollNumber, name, marks);
        students.add(student);
        
        System.out.println("\nStudent result added successfully!");
        System.out.println(student);
    }
    
    private static void viewAllStudents() {
        if (students.isEmpty()) {
            System.out.println("\nNo student records found!");
            return;
        }
        
        System.out.println("\n--- All Student Results ---");
        for (Student student : students) {
            System.out.println(student);
        }
    }
    
    private static void searchStudent() {
        if (students.isEmpty()) {
            System.out.println("\nNo student records found!");
            return;
        }
        
        System.out.print("\nEnter Roll Number to search: ");
        String rollNumber = scanner.nextLine();
        
        boolean found = false;
        for (Student student : students) {
            if (student.getRollNumber().equalsIgnoreCase(rollNumber)) {
                System.out.println("\n--- Student Found ---");
                System.out.println(student);
                found = true;
                break;
            }
        }
        
        if (!found) {
            System.out.println("\nNo student found with roll number: " + rollNumber);
        }
    }
    
    private static int getIntInput(String prompt, int min, int max) {
        while (true) {
            try {
                System.out.print(prompt);
                int value = Integer.parseInt(scanner.nextLine());
                if (value >= min && value <= max) {
                    return value;
                }
                System.out.printf("Please enter a number between %d and %d.\n", min, max);
            } catch (NumberFormatException e) {
                System.out.println("Invalid input. Please enter a valid number.");
            }
        }
    }
}
